### PR TITLE
File Server Integration Tests using Rest Assured

### PR DESCRIPTION
**Pull request by DivyaSreeMunagavalasa**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -25,34 +25,34 @@ pipeline {
...
 
-    stage('Unit Tests and CodeCoverage Test'){
+    stage('Unit Tests and Code Coverage Test'){
       steps{
         script{
           sh 'docker compose -f docker-compose.test.yml up test'
         }
         xunit (
-          thresholds: [ skipped(failureThreshold: '0'), failed(failureThreshold: '0') ],
+          thresholds: [ skipped(failureThreshold: '1'), failed(failureThreshold: '2') ],
```
Comment (pranavrd) [inline Jenkinsfile:34]: keep it 0 and 0

File: Jenkinsfile [hunk 2]
```diff
@@ -84,24 +84,34 @@ pipeline {
...
-            startZap ([host: 'localhost', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
-              sh 'curl http://127.0.0.1:8090/JSON/pscan/action/disableScanners/?ids=10096'
-              sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/fs/Newman/iudx-file-server-api.Release-v5.0.0.postman_collection.json -e /home/ubuntu/configs/fs-postman-env.json -n 2 --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/fs/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
-            runZapAttack()
+            startZap ([host: '0.0.0.0', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
+            sh 'curl http://0.0.0.0:8090/JSON/pscan/action/disableScanners/?ids=10096'
           }
         }
+        script{
+            sh 'scp /home/ubuntu/configs/rs-config-test.json ./configs/config-test.json'
```
Comment (pranavrd) [inline Jenkinsfile:96]: config would be `fs-config-test.json` not `rs-config-test.json`

File: src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            baseURI = "http://localhost";
+        }
+
+        String testPort = config.getString("httpPort");
+        // String testPort = System.getProperty("intTestPort");
+        if (testPort != null) {
+            port = Integer.parseInt(testPort);
+        } else {
+            port = 8443;
+        }
```
Comment (pranavrd) [inline src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java:47]: Need to get the testOnDepl environment variable for integration tests on deployment

File: src/test/java/iudx/file/server/apiserver/integrationtests/files/FileServerIntegrationTests.java [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+
+    private static String sampleFileId;
+    private static String archiveFileId;
+    private static String externalStorageFileId;
+    String invalidFileId = "_abced";
+    String nonExistingArchiveId ="83c2e5c2-3574-4e11-9530-2b1fbdfce832/8185010f-705d-4966-ac44-2050887c68f3_invalid.txt";
+
+    boolean isSample=true;
+    String invalidToken ="abc";
+    String fileDownloadURL = "https://docs.google.com/document/d/19f6oOIxHVjC3twcRHQATjrEXJsDO0rLixoFgLV7xMxk/edit?usp=sharing";
```
Comment (pranavrd) [inline src/test/java/iudx/file/server/apiserver/integrationtests/files/FileServerIntegrationTests.java:59]: Get these values from configs

File: src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java (hunk 3 above)
Comment (DivyaSreeMunagavalasa) [reply src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java]: Added this part in RestAssuredConfiguration file:

 boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }

Comment (DivyaSreeMunagavalasa) [reply src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java]: boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }


Added this part in RestAssuredConfiguration file.

Comment (DivyaSreeMunagavalasa) [reply src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java]: boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }

File: src/test/java/iudx/file/server/apiserver/integrationtests/files/FileServerIntegrationTests.java (hunk 4 above)
Comment (DivyaSreeMunagavalasa) [reply src/test/java/iudx/file/server/apiserver/integrationtests/files/FileServerIntegrationTests.java]: Fetched these values from config file.

File: Jenkinsfile (hunk 2 above)
Comment (DivyaSreeMunagavalasa) [reply Jenkinsfile]: Done.

File: Jenkinsfile (hunk 1 above)
Comment (DivyaSreeMunagavalasa) [reply Jenkinsfile]: Done

**Comment by pranavrd** [comment]
ok to test

**Comment by pranavrd** [comment]
ok to test

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

File: Jenkinsfile [hunk 5]
```diff
@@ -67,11 +67,10 @@ pipeline {
       }
     }
 
-    stage('Start File server for Integration Tests'){
+    stage('Start File-Server for Performance and Integration Testing'){
       steps{
         script{
-            sh 'scp src/test/resources/iudx-file-server-api.Release-v5.0.0.postman_collection.json jenkins@jenkins-master:/var/lib/jenkins/iudx/fs/Newman/'
-            sh 'docker compose -f docker-compose.test.yml up -d integTest'
+            sh 'docker compose -f docker-compose.test.yml up -d perfTest'
```
Comment (karun-singh) [inline Jenkinsfile:73]: The service name defined in the compose is integTest not perfTest
```suggestion
            sh 'docker compose -f docker-compose.test.yml up -d integTest'
```

File: Jenkinsfile [hunk 6]
```diff
@@ -67,11 +67,10 @@ pipeline {
       }
     }
 
-    stage('Start File server for Integration Tests'){
+    stage('Start File-Server for Performance and Integration Testing'){
```
Comment (karun-singh) [inline Jenkinsfile:70]: There are no Jmeter performance tests being done.
```suggestion
    stage('Start File-Server for Integration Tests'){
```

File: Jenkinsfile [hunk 7]
```diff
@@ -84,24 +83,34 @@ pipeline {
...
-              sh 'curl http://127.0.0.1:8090/JSON/pscan/action/disableScanners/?ids=10096'
-              sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/fs/Newman/iudx-file-server-api.Release-v5.0.0.postman_collection.json -e /home/ubuntu/configs/fs-postman-env.json -n 2 --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/fs/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
-            runZapAttack()
+            startZap ([host: '0.0.0.0', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
+            sh 'curl http://0.0.0.0:8090/JSON/pscan/action/disableScanners/?ids=10096'
           }
         }
+        script{
+            sh 'scp /home/ubuntu/configs/fs-config-test.json ./configs/config-test.json'
+            sh 'mvn test-compile failsafe:integration-test -DskipUnitTests=true -DintTestProxyHost=jenkins-master-priv -DintTestProxyPort=8090 -DintTestHost=jenkins-slave1 -DintTestPort=8080'
```
Comment (karun-singh) [inline Jenkinsfile:96]: The test port for fs would be 8443, as configured in ci config. 
```suggestion
            sh 'mvn test-compile failsafe:integration-test -DskipUnitTests=true -DintTestProxyHost=jenkins-master-priv -DintTestProxyPort=8090 -DintTestHost=jenkins-slave1 -DintTestPort=8443'
```

**Review by karun-singh — CHANGES_REQUESTED** [review]

**Comment by karun-singh** [comment]
retest this please

File: Jenkinsfile [hunk 8]
```diff
@@ -84,24 +83,34 @@ pipeline {
...
-            startZap ([host: 'localhost', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
-              sh 'curl http://127.0.0.1:8090/JSON/pscan/action/disableScanners/?ids=10096'
-              sh 'HTTP_PROXY=\'127.0.0.1:8090\' newman run /var/lib/jenkins/iudx/fs/Newman/iudx-file-server-api.Release-v5.0.0.postman_collection.json -e /home/ubuntu/configs/fs-postman-env.json -n 2 --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/fs/Newman/report/report.html --reporter-htmlextra-skipSensitiveData'
-            runZapAttack()
+            startZap ([host: '0.0.0.0', port: 8090, zapHome: '/var/lib/jenkins/tools/com.cloudbees.jenkins.plugins.customtools.CustomTool/OWASP_ZAP/ZAP_2.11.0'])
+            sh 'curl http://0.0.0.0:8090/JSON/pscan/action/disableScanners/?ids=10096'
           }
         }
+        script{
+            sh 'scp /home/ubuntu/configs/fs-config-test.json ./configs/config-test.json'
```
Comment (karun-singh) [inline Jenkinsfile:95]: There is no configs directory in git repo. Need to create one in the workspace during pipeline execution.
```suggestion
            sh 'mkdir configs; scp /home/ubuntu/configs/fs-config-test.json ./configs/config-test.json'
```

**Review by karun-singh — CHANGES_REQUESTED** [review]

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Comment by karun-singh** [comment]
retest this please

**Comment by pranavrd** [comment]
ok to test

**Review by pranavrd — APPROVED** [review]

**Review by karun-singh — APPROVED** [review]

**Pull request merged**